gm_dotnet_managed/GmodNET/GameConsoleWriter.cs

```diff
@@ -0,0 +1,20 @@
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+
+/// <summary>
+/// Redirects Console's output to game's console
+/// </summary>
+public class GameConsoleWriter : TextWriter
+{
+    public override string NewLine { get => "\n"; }
+    public override Encoding Encoding => Encoding.UTF8;
+
+    [DllImport("tier0", CallingConvention = CallingConvention.Cdecl)]
+    private static extern void Msg([MarshalAs(UnmanagedType.LPUTF8Str)] string msg);
+
+    public override void Write(string value)
+    {
+        Msg(value);
```

Comment:
I have two questions here:

1) What happens if several threads try to write to console simultaneously? Maybe we should acquire lock on `this` to ensure thread safety?  

2) There is a note in Source SDK documentation (https://developer.valvesoftware.com/wiki/Developer_Console_Control):

> Only 989 characters can be output to the console at once!

What happens if we violate this limit? Should we take care of this in C# code?

Reply:
1. only thing not working in multi threading is broken new lines, because `TextWriter.Write` called seperately 
```cs
// WriteLine(string value)
TextWriter.Write(value);
TextWriter.Write(TextWriter.NewLine);
```
i'll fix it
2. i don't know, will test it now

Reply:
1. fixed
2. lie